### PR TITLE
Registry teardown

### PR DESCRIPTION
**Pull request by felixwang9817**
**What this PR does / why we need it**: As of right now, registries do not get cleaned up during teardown. This is especially bad for GCS and S3 registries. This PR ensures that registries get cleaned up during teardown. 

**Which issue(s) this PR fixes**:

Fixes #

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: sdk/python/feast/registry.py [hunk 1]
```diff
@@ -610,6 +641,10 @@ def update_registry_proto(self, registry_proto: RegistryProto):
         self._write_registry(registry_proto)
         return
 
+    def teardown(self):
+        self.s3_client.Object(self._bucket, self._key).delete()
+        return
```
Comment (achals) [inline sdk/python/feast/registry.py:646]: nit, returns aren't necessary for methods that don't return anything

**Review by achals — APPROVED** [review]

File: sdk/python/feast/registry.py [hunk 2]
```diff
@@ -21,6 +21,8 @@
 from typing import List, Optional
 from urllib.parse import urlparse
 
+from google.cloud.exceptions import NotFound
```
Comment (woop) [inline sdk/python/feast/registry.py:24]: Should this dependency be eagerly loaded? What about teams that don't use GCP?

Comment (felixwang9817) [reply sdk/python/feast/registry.py:24]: Good catch - I'll place this import statement in the method.

File: sdk/python/feast/registry.py [hunk 3]
```diff
@@ -399,6 +395,10 @@ def refresh(self):
         """Refreshes the state of the registry cache by fetching the registry state from the remote registry store."""
         self._get_registry_proto(allow_cache=False)
 
+    def teardown(self):
+        """Tears down all local and cloud resources."""
```
Comment (woop) [inline sdk/python/feast/registry.py:399]: Why does the registry tear down all local and cloud resources? Seems confusing. Wouldn't we only be removing all registry specific resources?

Comment (felixwang9817) [reply sdk/python/feast/registry.py:399]: Hm, `deployed` doesn't sound right to me - it sounds like there's a server involved. What about just `Tears down (removes) the registry.`?

Comment (woop) [reply sdk/python/feast/registry.py:399]: sgtm

Comment (woop) [reply sdk/python/feast/registry.py:399]: I just want to be clear that we aren't touching any real infrastructure. We are dealing with the registry here.

**Review by achals — APPROVED** [review]
/lgtm

**Pull request merged**